src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -583,6 +545,20 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
                     break;
             }
 
+            TypeSymbol resultType = signature.ReturnType;
+            BoundExpression resultLeft = left;
+            BoundExpression resultRight = right;
+            ConstantValue resultConstant = null;
+
+            if ((object)signature.LeftType != null)
+            {
+                Debug.Assert((object)signature.RightType != null);
+
+                resultLeft = GenerateConversionForAssignment(signature.LeftType, left, diagnostics);
```

Comment:
>GenerateConversionForAssignment [](start = 29, length = 31)

It is hard to say if we are going to have exactly the same behavior around conversions. I think we should duplicate previous logic. It shouldn't be too hard because ```best``` is readily available from  BindSimpleBinaryOperatorParts and we can check whether ```foundOperator``` and operator is not for NullableNull, otherwise, it looks like we don't need to perform this conversion.